src/library.c

```diff
@@ -0,0 +1,851 @@
+/*
+ * Copyright (C) 2015 Christian Meffert <christian.meffert@googlemail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+
+#ifdef HAVE_CONFIG_H
+# include <config.h>
+#endif
+
+#include "library.h"
+
+#include <errno.h>
+#include <event2/event.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <pthread.h>
+#ifdef HAVE_PTHREAD_NP_H
+# include <pthread_np.h>
+#endif
+#include <stdlib.h>
+#include <string.h>
+#include <unictype.h>
+#include <uninorm.h>
+#include <unistr.h>
+
+#include "cache.h"
+#include "commands.h"
+#include "conffile.h"
+#include "db.h"
+#include "library/filescanner.h"
```

Comment:
Ok, sounds sensible. The inputs I'm working on is just focused on getting audio and metadata from the stream to the player, so right now it is not a suitable owner of the scan. Anyway, I agree, let's start by getting this up and running, then we can consider later how to make it more generic.